python/dgl/distributed/rpc_client.py

```diff
@@ -183,30 +186,69 @@ def finalize_client():
     """Release resources of this client."""
     rpc.finalize_sender()
     rpc.finalize_receiver()
+    if SAMPLER_POOL is not None:
+        SAMPLER_POOL.close()
+        SAMPLER_POOL.join()
     global INITIALIZED
     INITIALIZED = False
 
+
 def shutdown_servers():
     """Issue commands to remote servers to shut them down.
 
     Raises
     ------
     ConnectionError : If anything wrong with the connection.
     """
-    if rpc.get_rank() == 0: # Only client_0 issue this command
+    if rpc.get_rank() == 0:  # Only client_0 issue this command
         req = rpc.ShutDownRequest(rpc.get_rank())
         for server_id in range(rpc.get_num_server()):
             rpc.send_request(server_id, req)
 
+SAMPLER_POOL = None
+NUM_SAMPLER_WORKERS = 0
+
+
+def _init_rpc(ip_config, max_queue_size, net_type, role):
+    ''' This init function is called in the worker processes.
+    '''
+    try:
+        connect_to_server(ip_config, max_queue_size, net_type)
+        init_kvstore(ip_config, role)
+    except Exception as e:
+        print(e, flush=True)
+        traceback.print_exc()
+        raise e
+
+def get_sampler_pool():
+    """Return the sampler pool and num_workers"""
+    return SAMPLER_POOL, NUM_SAMPLER_WORKERS
+
+
+def init_rpc(ip_config, num_workers=0, max_queue_size=MAX_QUEUE_SIZE, net_type='socket'):
```

Comment:
let's rename it as `initialize`.

should we move the code here to `__init__.py`? so we don't need to import kvstore in rpc_client.py? The initialization now also include kvstore.

Reply:
we should also call `reset` in initialization.

Reply:
Moved to dist_context.py